src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,204 @@
+        this.squarelotron = squarelotron;
+    }
+
+    /**
+     * Constructor Sqarelotron.
+     *
+     * @param n - Size of squarelotron array.
+     */
+    public Squarelotron(final int n) {
+        this.size = n;
+        this.squarelotron = new int[size][size];
+
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                squarelotron[i][j] = size * i + j + 1;
+            }
+        }
+    }
+
+    /**
+     * Gets the maximum number of ring in the squarelotron.
+     *
+     * @return Numbers maximum of rings.
+     */
+    public int getMaxRings() {
+        return (this.size % 2 == 0) ? size / 2 : size / 2 + 1;
+    }
+    /**
+      * Modifies A Upside-Down Flip of the outer ring of Squarelotron.
+      *
+      * @param ring - Number of ring.
+      * @return Instance new Squarelotron.
+      */
+    public Squarelotron upsideDownFlip(int ring) {
+
+        if (ring > this.getMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotron = new Squarelotron(this.size);
+
+        int first = ring - 1;
+        int last = size - ring;
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+                if ((i == first || i == last) && (j >= first && j <= last)) {
+                        squarelotron.squarelotron[i][j] = this.squarelotron[size - 1 - i][j];
+                }
+
+                if ((i > first && i < last) && (j == first || j == last)) {
+                        squarelotron.squarelotron[i][j] = this.squarelotron[size - 1 - i][j];
+                }
+            }
+        }
+        return squarelotron;
+    }
+
+    /**
+     * Modifies the matrix - move the number diagonally in the Matrix.
+     *
+     * @param ring - Number the ring affected.
+     * @return New instance of Squarelotron
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+
+        if (ring > this.getMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotron = new Squarelotron(this.size);
+
+        int first = ring - 1; // 0
+        int last = size - ring; // 3
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+
+                if ((i == first || i == last) && (j >= first && j <= last)) {
+                        squarelotron.squarelotron[j][i] = this.squarelotron[i][j];
+                }
+
+                if ((i > first && i < last) && (j == first || j == last)) {
+                        squarelotron.squarelotron[j][i] = this.squarelotron[i][j];
+                }
+            }
+        }
+        return squarelotron;
+    }
+
+    /**
+     * Rotates Matrix 90 grades positive or negative position.
+     *
+     * @param numberOfTurns Number or rotations.
+     */
+    public void rotateRight(int numberOfTurns) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 71 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)